core/src/main/scala/akka/persistence/r2dbc/internal/PubSub.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.actor.typed.pubsub.Topic
+import akka.annotation.InternalApi
+import akka.persistence.Persistence
+import akka.persistence.PersistentRepr
+import akka.persistence.query.typed.EventEnvelope
+import akka.persistence.r2dbc.query.TimestampOffset
+import akka.persistence.typed.PersistenceId
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object PubSub extends ExtensionId[PubSub] {
+  def createExtension(system: ActorSystem[_]): PubSub = new PubSub(system)
+
+  // Java API
+  def get(system: ActorSystem[_]): PubSub = apply(system)
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class PubSub(system: ActorSystem[_]) extends Extension {
+  private val topics = new ConcurrentHashMap[String, ActorRef[Any]]
```

Comment:
Could it not be `ActorRef[Topic.Command[EventEnvelope[Event]]]` here already and skip the unsafeUpcast.narrow below?

Reply:
yes, it can have some more type, but Event is unknown

Reply:
Ofc, I missed that it is a method type param.